components/calendar/src/provider.rs

```diff
@@ -151,3 +152,389 @@ pub struct WeekDataV1 {
     /// For a given week, the minimum number of that week's days present in a given month or year for the week to be considered part of that month or year.
     pub min_week_days: u8,
 }
+
+/// Bitset representing weekdays that are part of the 'weekend'.
+///
+/// Returns days via [`Self::days()`] method which returns [WeekendDays], an [Iterator].
+/// Users of this data do not need to interact with this Bitset,
+/// you can access [WeekendDays] using [`WeekDataV2::weekend()`].
+///
+/// # Internal representation
+///
+/// This Bitset uses an [u8] to represent the weekend, thus leaving one bit free.
+/// Each bit represents a day in the following order:
+///
+///   ┌▷Mon
+///   │┌▷Tue
+///   ││┌▷Wed
+///   │││┌▷Thu
+///   ││││ ┌▷Fri
+///   ││││ │┌▷Sat
+///   ││││ ││┌▷Sun
+///   ││││ │││
+/// 0b0000_1010
+///
+/// Please note that this is not a range, this are the discrete days representing a weekend. Other examples:
+/// 0b0101_1000 -> Tue, Thu, Fri
+/// 0b0000_0110 -> Sat, Sun
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct WeekendSet(u8);
+
+#[cfg(feature = "datagen")]
+impl databake::Bake for WeekendSet {
+    fn bake(&self, ctx: &databake::CrateEnv) -> databake::TokenStream {
+        ctx.insert("icu_calendar");
+        let bitset = self.0.bake(ctx);
+        databake::quote! {
+            icu_calendar::provider::WeekendSet::from_unvalidated_bitset(#bitset)
+        }
+    }
+}
+
+impl WeekendSet {
+    /// Creates a new [WeekendSet] using the two provided days.
+    ///
+    /// Because of the current [CLDR spec](https://www.unicode.org/reports/tr35/tr35-dates.html#Date_Patterns_Week_Elements),
+    /// these are not a range, these are the two days representing a weekend.
+    /// If days are equal, weekend only has one day.
+    pub fn new(first_day: IsoWeekday, second_day: IsoWeekday) -> Self {
+        WeekendSet(first_day.bit_value() | second_day.bit_value())
+    }
+
+    /// Creates a new [WeekendSet] from an unvalidated bitset. Used from compiled data.
+    pub const fn from_unvalidated_bitset(bitset: u8) -> Self {
```

Comment:
The `new` constructor looks like it could be `const`

Reply:
It could, but the way I understand it is that we need a way to [directly set the u8](https://github.com/JhonnyBillM/icu4x/blob/e3b3aed577197cbf361022ffccf632e3d7692855/components/calendar/src/provider.rs#L187), which is the only field in the struct.

Please let me know if I am missing something on how this should work

Reply:
You shouldn't directly set the `u8` in an unchecked way when you can go through the constructor. Bake should produce something like

```rust
icu_calendar::provider::WeekendSet::new(IsoWeekday::Saturday, IsoWeekday::Sunday)
```